### PR TITLE
feat(UI-1047): add hubspot connection

### PR DESCRIPTION
**Pull request by J1za**
## Description
Based on this form:

https://github.com/autokitteh/autokitteh/blob/ee626d8aa29d5bd87739f4012b6d39924ed5a8db/web/static/hubspot/connect/index.html

Please create a new/edit connection forms on our side.

Take the icon from here and mention the source in the index.ts file:

https://www.svgrepo.com/svg/331433/hubspot
## Linear Ticket
https://linear.app/autokitteh/issue/UI-1047/add-hubspot-connection
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/enums/components/connection.enum.ts [hunk 1]
```diff
@@ -153,6 +154,11 @@ export const IntegrationsMap: Record<Integrations, IntegrationSelectOption> = {
 		label: "Twilio",
 		value: Integrations.twilio,
 	},
+	hubspot: {
+		icon: TwilioIcon,
```
Comment (RonenMars) [inline src/enums/components/connection.enum.ts:158]: HubSpot icon

Comment (RonenMars) [inline src/enums/components/connection.enum.ts:158]:
```suggestion
		icon: HubSpotIcon,
```

File: src/locales/en/errors/translation.json [hunk 2]
```diff
@@ -91,5 +91,6 @@
 	"errorFetchingEvent": "Couldn't fetch event",
 	"eventNotFound": "Event details not found",
 	"eventNotFoundExtended": "Event details not found, {{eventId}} - error: {{error}}",
-	"fetchSessionFailed": "Failed to fetch session logs"
+	"fetchSessionFailed": "Failed to fetch session logs",
+	"errorFetchingIntegrations": "Error fetching integrations"
```
Comment (RonenMars) [inline src/locales/en/errors/translation.json:95]: Where do you use it?

File: src/locales/en/services/translation.json [hunk 3]
```diff
@@ -90,6 +90,7 @@
 	"tokenCreationError": "Error occurred during the token creation",
 	"deploymentCreatedSuccessfully": "Deployment created with id: {{deploymentId}}",
 	"failedFetchingLogRecordsBySessionId": "Failed to fetch log records by session ID: {{sessionId}}, error: {{error}}, page token: {{pageToken}}, page size: {{pageSize}}, log type: {{logType}}",
-	"connection": "Connection"
+	"connection": "Connection",
+	"errorFetchingIntegrationsForConnectionsExtended": "Error fetching integrations for connections, connection ID: {{connectionId}}, error: {{error}}"
```
Comment (RonenMars) [inline src/locales/en/services/translation.json:94]: Where do you use it?

File: src/locales/en/errors/translation.json (hunk 2 above)
Comment (J1za) [reply src/locales/en/errors/translation.json:95]: 1. inside SessionViewer component
2. inside useCacheStore

File: src/locales/en/services/translation.json (hunk 3 above)
Comment (J1za) [reply src/locales/en/services/translation.json:94]: inside connection.service for `get` function

File: src/enums/components/connection.enum.ts (hunk 1 above)
Comment (J1za) [reply src/enums/components/connection.enum.ts]: changed

File: src/assets/image/icons/connections/index.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ export { default as GoogleGmailIcon } from "@assets/image/icons/connections/Goog
 export { default as GoogleSheetsIcon } from "@assets/image/icons/connections/GoogleSheets.svg?react";
 export { default as GrpcIcon } from "@assets/image/icons/connections/Grpc.svg?react";
 export { default as HttpIcon } from "@assets/image/icons/connections/Http.svg?react";
+export { default as HubspotIcon } from "@assets/image/icons/connections/Hubspot.svg?react";
```
Comment (RonenMars) [inline src/assets/image/icons/connections/index.ts:16]: @J1za please add a link for the icon source 🙏🏻

File: src/components/organisms/connections/integrations/integrationEditForm.tsx [hunk 5]
```diff
@@ -106,7 +106,11 @@ export const IntegrationEditForm = ({
 
 	useEffect(() => {
 		if (connectionVariables) {
-			setFormValues(connectionVariables, integrationVariablesMapping[integrationType], setValue);
+			setFormValues(
+				connectionVariables,
+				integrationVariablesMapping[integrationType as keyof typeof integrationVariablesMapping],
+				setValue
+			);
```
Comment (RonenMars) [inline src/components/organisms/connections/integrations/integrationEditForm.tsx:113]: Why do we change it in this PR?

File: src/assets/image/icons/connections/index.ts (hunk 4 above)
Comment (J1za) [reply src/assets/image/icons/connections/index.ts:16]: Its added inside svg file

File: src/components/organisms/connections/integrations/integrationEditForm.tsx (hunk 5 above)
Comment (J1za) [reply src/components/organisms/connections/integrations/integrationEditForm.tsx:113]: I had ts error

File: src/assets/image/icons/connections/index.ts (hunk 4 above)
Comment (RonenMars) [reply src/assets/image/icons/connections/index.ts:16]: Please add inside the index.ts

Comment (RonenMars) [reply src/assets/image/icons/connections/index.ts:16]: There are many variations in svgrepo.com, we better save the full link to the icon, in order to be able to see the license from there easily.

File: src/components/organisms/connections/integrations/integrationEditForm.tsx (hunk 5 above)
Comment (RonenMars) [reply src/components/organisms/connections/integrations/integrationEditForm.tsx:113]: Ok

File: src/assets/image/icons/connections/index.ts (hunk 4 above)
Comment (J1za) [reply src/assets/image/icons/connections/index.ts:16]: added

**Comment by RonenMars** [comment]
Please use this icon to get MIT license and not another: https://www.svgrepo.com/svg/443148/brand-hubspot

File: src/assets/image/icons/connections/index.ts (hunk 4 above)
Comment (RonenMars) [reply src/assets/image/icons/connections/index.ts:17]: Please change the icon to this one:
https://www.svgrepo.com/svg/443148/brand-hubspot

**Review by RonenMars — APPROVED** [review]

File: src/assets/image/icons/connections/index.ts (hunk 4 above)
Comment (J1za) [reply src/assets/image/icons/connections/index.ts:17]: changed

Comment (RonenMars) [reply src/assets/image/icons/connections/index.ts:17]: Thanks!

**Pull request merged**